### ISSUE TITLE
SDK command endpoint only accepts [Core]Commands

### ISSUE THREAD

**Issue by tonyespy** [edgexfoundry/device-sdk-go#227]
While working on core-contracts [issue #27](https://github.com/edgexfoundry/go-mod-core-contracts/issues/27) (remove device profile from device), I wanted to check the names of the readings being returned when calling the command endpoint with certain resource names (aka what we're now going to call device-commands).  What I discovered is that the SDK validates the incoming command name against the list of "commands" present in the device profile. This behavior is different than the original Java SDK, and I from what I've been told the new device-sdk-c as well. See the following for reference on how this is done:

 * https://github.com/edgexfoundry/device-sdk-go/blob/master/internal/handler/command.go#L56  
 * https://github.com/edgexfoundry/device-sdk-go/blob/master/internal/cache/profiles.go#L158
 * https://github.com/edgexfoundry/device-sdk-go/blob/master/internal/cache/profiles.go#L73

I checked and this was changed in the major refactoring of device-sdk-go done last October. As we have no black box tests, this wasn't caught...

To reproduce:
* make sure curl and jq are both installed
* install the edgex snap (sudo snap install edgexfoundry --edge)
* edit the file /var/snap/edgexfoundry/config/device-random/res/device.random.yaml and delete the first three commands in the command section. The only remaining command should be GenerateRandomValue_Int32
* enable the device-random service (sudo snap set edgexfoundry device-random=on)
* run the the following curl commands; the second two commands fail, and the last will succeed (for **GenerateRandomValue_Int8**) will succeed

```
$ id=`curl -s http://localhost:48081/api/v1/device/name/Random-Integer-Generator01 | jq '.id' | tr -d '"'`

$ curl -s "http://localhost:49988/api/v1/device/$id/GenerateRandomValue_Int8" 
$ curl -s "http://localhost:49988/api/v1/device/$id/GenerateRandomValue_Int16"
$ curl -s "http://localhost:49988/api/v1/device/$id/GenerateRandomValue_Int32" 
```

**Comment by tonyespy** [comment]
@iain-anderson Can you please review?

**Comment by iain-anderson** [comment]
Suggest this is fixed as part of #218 - probably by removing the use of CommandExists().

**Comment by tonyespy** [comment]
I think it's related to #218, but not solely resolved. Let's discuss at our upcoming meeting.
